src/app/api/database/[database_id]/user/route.ts

```diff
@@ -0,0 +1,183 @@
+import { and, eq } from "drizzle-orm";
+import { NextResponse } from "next/server";
+import withDatabaseOperation, {
+  DatabasePermission,
+} from "@/lib/with-database-ops";
+import { database_user_role, database_role } from "@/db/schema";
+import { db } from "@/db";
+import zod from "zod";
+
+export const runtime = "edge";
+
+const assignUserSchema = zod.object({
+  user: zod.string(),
+  role: zod.string(),
+});
+
+const deleteUserSchema = zod.object({
+  user: zod.string(),
+});
+
+function checkOwnerPermission(permission: DatabasePermission) {
+  if (!permission.isOwner) {
+    return NextResponse.json(
+      {
+        success: false,
+        error: "You do not have permission",
+      },
+      { status: 403 }
+    );
+  }
+}
+
+async function getUserRoleInfo(userId: string, databaseId: string) {
+  return db.query.database_user_role.findFirst({
+    where: and(
+      eq(database_user_role.databaseId, databaseId),
+      eq(database_user_role.userId, userId)
+    ),
+    columns: {
+      roleId: true,
+    },
+  });
+}
+
+async function checkUserHasOwnerRole(roleId: string) {
+  const roleInfo = await db.query.database_role.findFirst({
+    where: eq(database_role.id, roleId),
+    columns: {
+      isOwner: true,
+    },
+  });
+  return roleInfo?.isOwner;
+}
+
+export const POST = withDatabaseOperation(
+  async ({ user, body, database, permission }) => {
+    const parsed = assignUserSchema.safeParse(body);
+
+    if (!parsed.success) {
+      return NextResponse.json(
+        {
+          success: false,
+          error: "Invalid request body",
+        },
+        { status: 400 }
+      );
+    }
+
+    const permissionError = checkOwnerPermission(permission);
```

Comment:
You don't need. Because `permission.isOwner` always tell if the current user is in the owner group.